aip/cloud/25146.md

```diff
@@ -0,0 +1,25 @@
+---
+id: 25146
+state: draft
+created: TBD
+placement:
+  category: gcp
+---
+
+# Generic Fields
+
+Guidelines related to generic fields in [AIP-146](../0146.md) **should** be applied
+with the following exceptions.
+
+## Maps
+
+Fields of type `map` **must not** be composed of type Message. `map<string, string>`
+is permissible, but `map<string, google.protobuf.Struct>` is not allowed.
```

Comment:
What about other `map<primitive, primitive>` combos? We will certainly get that question.

Reply:
I think `map<string, primitive>` should be fine, but I believe in JSON the key must always be a string

Reply:
According to the [protobuf JSON schema](https://developers.google.com/protocol-buffers/docs/proto3#json), "All keys are converted to strings", so it is still _technically_ valid to have other, non-string primitives used as keys. Not saying it's common or a good idea, but the DSL and encoding spec say it is allowed, so we should at least focus the guidance to clarify specifically what we are looking for and what is not allowed, because there is always one service that tests the boundaries :)

Reply:
If the keys end up as strings in the API then clients will only interact with them as strings, so no problems there!

I'll change it to say `map<primitive, primitive>` is fine